pkg/cache/v3/simple.go

```diff
@@ -306,6 +306,39 @@ func (cache *snapshotCache) CreateWatch(request *Request, value chan Response) f
 	snapshot, exists := cache.snapshots[nodeID]
 	version := snapshot.GetVersion(request.TypeUrl)
 
+	if exists && knownResourceNames != nil {
+		diff := []string{}
+		for _, r := range request.ResourceNames {
+			if _, ok := knownResourceNames[r]; !ok {
+				diff = append(diff, r)
+			}
+		}
+		if cache.log != nil {
+			cache.log.Debugf("nodeID %q requested %s%v and known %v. Diff %v", nodeID,
+				request.TypeUrl, request.ResourceNames, knownResourceNames, diff)
+		}
+		if len(diff) > 0 {
+			found := false
+			if cache.log != nil {
+				cache.log.Debugf("nodeID %q still needs %v", nodeID, diff)
+			}
+
+			resources := snapshot.GetResourcesAndTTL(request.TypeUrl)
+			for _, name := range diff {
+				if _, exists := resources[name]; exists {
```

Comment:
So I think this will yield unwanted results if we enter a case where resources have the same name but different objects. That's one of the reasons we did the hash map in incremental so we handle that scenario. I'd be okay with moving a field inside of the streamState to something like `State map[string]struct{} // current state of resources in SOTW` where the `struct{}` is the value we can compare. That way we guarantee we won't falsely match

Reply:
Sorry, not sure if I follow again. What do you mean in `a case where resources have the same name but different objects.` ? 

The logic I added there checks if newly requested resource is already known to control plane.

Reply:
Ah sorry I think I misunderstood, just tracking the resource name is sufficient for creating a diff in SOTW. I was in the incremental xDS mindset where we'd also need to create versions for the new resource. Ignore my previous statement